[GStreamer][MediaStream] Apply timestamps to mock audio/video buffers

philn opened the pull request:
#### 4f577536c5c008829f38ca9f270ade55a615c52a
<pre>
[GStreamer][MediaStream] Apply timestamps to mock audio/video buffers
<a href="https://bugs.webkit.org/show_bug.cgi?id=243399">https://bugs.webkit.org/show_bug.cgi?id=243399</a>

Reviewed by Xabier Rodriguez-Calvar.

In case the buffers are pushed to an encoder, valid timestamps are required.

* Source/WebCore/platform/graphics/gstreamer/VideoFrameGStreamer.cpp:
(WebCore::VideoFrameGStreamer::createFromPixelBuffer):
* Source/WebCore/platform/mediastream/gstreamer/MockRealtimeAudioSourceGStreamer.cpp:
(WebCore::MockRealtimeAudioSourceGStreamer::render):
* Source/WebCore/platform/mediastream/gstreamer/MockRealtimeVideoSourceGStreamer.cpp:
(WebCore::MockRealtimeVideoSourceGStreamer::updateSampleBuffer):

Canonical link: <a href="https://commits.webkit.org/253106@main">https://commits.webkit.org/253106@main</a>
</pre>

calvaris commented on Source/WebCore/platform/mediastream/gstreamer/MockRealtimeAudioSourceGStreamer.cpp:
```diff
@@ -116,10 +116,13 @@ void MockRealtimeAudioSourceGStreamer::render(Seconds delta)
         totalFrameCount -= bipBopCount;
         frameCount = std::min(totalFrameCount, m_maximiumFrameCount);
 
+        MediaTime mediaTime((m_samplesRendered * G_USEC_PER_SEC) / sampleRate(), G_USEC_PER_SEC);
```
Better a more meaningful name than just decapitalizing the type, maybe `presentationTime`

philn committed:
[GStreamer][MediaStream] Apply timestamps to mock audio/video buffers
https://bugs.webkit.org/show_bug.cgi?id=243399

Reviewed by Xabier Rodriguez-Calvar.

In case the buffers are pushed to an encoder, valid timestamps are required.

* Source/WebCore/platform/graphics/gstreamer/VideoFrameGStreamer.cpp:
(WebCore::VideoFrameGStreamer::createFromPixelBuffer):
* Source/WebCore/platform/mediastream/gstreamer/MockRealtimeAudioSourceGStreamer.cpp:
(WebCore::MockRealtimeAudioSourceGStreamer::render):
* Source/WebCore/platform/mediastream/gstreamer/MockRealtimeVideoSourceGStreamer.cpp:
(WebCore::MockRealtimeVideoSourceGStreamer::updateSampleBuffer):

Canonical link: https://commits.webkit.org/253106@main
